src/core/execute-serialize.c

```diff
@@ -1735,6 +1735,11 @@ static int exec_context_serialize(const ExecContext *c, FILE *f) {
         if (r < 0)
                 return r;
 
+        assert(c->private_var_tmp >= 0 && c->private_var_tmp < _PRIVATE_TMP_MAX);
```

Comment:
no need for such specific knowledge in serialization code? it's validated multiple times elsewhere after all.

Reply:
This ensures private_var_tmp is explicitly set before invoking the executor.
Of course, the same check is done at several more places, so redundant. But this is just an assertion and we have so many trivial assertions...

Reply:
> But this is just an assertion and we have so many trivial assertions...

Sure, but currently the serialization code carries none of those. but it's fine either way.